mongoproxy/README.md

```diff
@@ -0,0 +1,58 @@
+# mongoproxy
```

Comment:
I don't feel strongly but we do have precedent of including mock servers in the `.evergreen` folder - should we move mongoproxy into `.evergreen`?  ex: `.evergreen/csfle` contains a lot of Python mock KMS servers